File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = prices[0], max = 0;
        for(int i = 1; i < prices.length; i++){
            if(prices[i] < l){
                l = prices[i];
            }
            max = Math.max(max, prices[i] - l);
        }
        return max;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        int len = prices.size();
        int stock = prices[0], profit = 0;

        for(int i = 1; i < len; i++){
            if(prices[i] < stock){
                stock = prices[i];
            }
            profit = max(profit, prices[i] - stock);
        }

        return profit;

    }
};